lib/elixir/test/elixir/kernel/errors_test.exs

```diff
@@ -37,6 +37,19 @@ defmodule Kernel.ErrorsTest do
       'Foo.:"\#{:bar}"'
   end
 
+  test :invalid_identifier do
+    assert_compile_fail SyntaxError, "nofile:1: unexpected token: foo@", 'foo@'
```

Comment:
To be clear, if we use unexpected token, I would expect a single token in the error message. :)